ports/rp2/mphalport.c

```diff
@@ -83,6 +84,25 @@ void tud_cdc_rx_cb(uint8_t itf) {
     }
 }
 
+#if MICROPY_HW_USB_CDC_1200BPS_TOUCH
+static mp_sched_node_t mp_bootloader_sched_node;
+STATIC void usbd_cdc_run_bootloader_task(mp_sched_node_t *node) {
+    mp_hal_delay_ms(250);
+    machine_bootloader(0, NULL);
+}
+
+void tud_cdc_line_state_cb(uint8_t itf, bool dtr, bool rts) {
+    if (dtr == false && rts == false) {
```

Comment:
maybe comment here, like in nrf

Reply:
Done. Since you commented on this, should this duplicate block be in a common file somewhere for ports that use tinyusb and need this feature ? If it's in `.c` file and it's linked it will just enable this feature because `tud_cdc_line_state_cb` is weak.

Reply:
Yes that would be good to put it somewhere common.  It can go in `shared/`, maybe `shared/runtime/tinyusb_helpers.c`?

Reply:
Moved to `shared/runtime/tinyusb_helpers.c`.